python/tests/test_pyvw.py

```diff
@@ -347,94 +381,158 @@ def check_error_raises(type, argument):
+        features=Feature(name="col_x", value="x"),
         df=df,
     )
-    first_line = conv.process_df()[0]
+    first_line = conv.convert_df()[0]
     assert first_line == "1 id_1| col_x:2"
 
 
-def test_feature_constant_column_with_empty_name():
+def test_constant_feature_value_with_empty_name():
     df = pd.DataFrame({"idx": ["id_1"], "y": [1], "x": [2]})
     conv = DFtoVW(
-        label=SimpleLabel(Col("y")),
-        tag=Col("idx"),
-        namespaces=Namespace([Feature(name="", value=2)]),
+        label=SimpleLabel("y"),
+        tag="idx",
+        features=Feature(name="", value=2, value_from_df=False),
         df=df,
     )
-    first_line = conv.process_df()[0]
+    first_line = conv.convert_df()[0]
     assert first_line == "1 id_1| :2"
 
 
-def test_feature_variable_column_name():
+def test_variable_feature_name():
     df = pd.DataFrame({"y": [1], "x": [2], "a": ["col_x"]})
     conv = DFtoVW(
-        label=SimpleLabel(Col("y")),
-        namespaces=Namespace(Feature(name=Col("a"), value=Col("x"))),
+        label=SimpleLabel("y"),
+        features=Feature(name="a", value="x", name_from_df=True),
         df=df,
     )
-    first_line = conv.process_df()[0]
+    first_line = conv.convert_df()[0]
     assert first_line == "1 | col_x:2"
 
 
-def test_multiple_lines_conversion():
+def test_multiple_lines():
     df = pd.DataFrame({"y": [1, -1], "x": [1, 2]})
-    conv = DFtoVW(
-        label=SimpleLabel(Col("y")),
-        namespaces=Namespace(Feature(value=Col("x"))),
-        df=df,
-    )
-    lines_list = conv.process_df()
+    conv = DFtoVW(label=SimpleLabel("y"), features=Feature(value="x"), df=df,)
+    lines_list = conv.convert_df()
     assert lines_list == ["1 | 1", "-1 | 2"]
 
 
-def test_multiple_namespaces():
+def test_multiple_named_namespaces():
     df = pd.DataFrame({"y": [1], "a": [2], "b": [3]})
     conv = DFtoVW(
         df=df,
-        label=SimpleLabel(Col("y")),
+        label=SimpleLabel("y"),
         namespaces=[
-            Namespace(name="FirstNameSpace", features=Feature(Col("a"))),
-            Namespace(name="DoubleIt", value=2, features=Feature(Col("b"))),
+            Namespace(name="FirstNameSpace", features=Feature("a")),
+            Namespace(name="DoubleIt", value=2, features=Feature("b")),
         ],
     )
-    first_line = conv.process_df()[0]
+    first_line = conv.convert_df()[0]
     assert first_line == "1 |FirstNameSpace 2 |DoubleIt:2 3"
 
 
-def test_without_target():
+def test_without_target_multiple_features():
     df = pd.DataFrame({"a": [2], "b": [3]})
+    conv = DFtoVW(df=df, features=[Feature(col) for col in ["a", "b"]])
+    first_line = conv.convert_df()[0]
+    assert first_line == "| 2 3"
+
+
+def test_multiclass():
+    df = pd.DataFrame({"a": [1], "b": [0.5], "c": ["x"]})
     conv = DFtoVW(
-        df=df, namespaces=Namespace([Feature(Col("a")), Feature(Col("b"))])
+        df=df, label=MultiClass(name="a", weight="b"), features=Feature("c")
```

Comment:
had a hard time finding this test. This makes me realize this single test file might be too much. Not related with this PR.